api/envoy/config/tap/v3/common.proto

```diff
@@ -39,6 +39,7 @@ message TapConfig {
   // Exactly one of :ref:`match <envoy_api_field_config.tap.v3.TapConfig.match>` and
   // :ref:`match_config <envoy_api_field_config.tap.v3.TapConfig.match_config>` must be set. If both
   // are set, the :ref:`match <envoy_api_field_config.tap.v3.TapConfig.match>` will be used.
+  // [#next-major-version: common.matcher.v3.MatchPredicate should be refactored to use type.matcher.v3.MatchPredicate.]
```

Comment:
This doc change for tap isn't really related? And we don't have `type.matcher.v3.MatchPredicate` (same for elsewhere), and that's not part of the new matcher API either.

Reply:
`type.matcher.v3.MatchPredicate` This is being introduced in this PR. We spoke to @snowp and the Unified matcher is going to take time.

I will handle the tech debt in the next,  separate PR while HeaderMatcher references are moved to StringMatcher based `type.Matcher.v3.HeaderMatcher` (which is also introduced in this PR).

Reply:
Sorry, I know that is what we said, but we are actively working on the new matching protos now. I think it's worth intersecting these efforts. Let's circle back next week and see where we are at.

Reply:
Okay